app/doi.py

```diff
@@ -0,0 +1,57 @@
+import os
+import json
+
+import requests
+from requests.exceptions import HTTPError
+
+import logging
+
+logger = logging.getLogger()
+
+
+def mint_doi(event, _context, _kwargs):
+    payload = event["body"]
+    try:
+        DATACITE_REPOSITORY_ID = os.environ["DATACITE_REPOSITORY_ID"]
+        DATACITE_PASSWORD = os.environ["DATACITE_PASSWORD"]
+        DATACITE_ENDPOINT = os.environ["DATACITE_ENDPOINT"]
+        DATACITE_PREFIX = os.environ["DATACITE_PREFIX"]
+    except KeyError as e:
+        message = (
+            "Garden server was unable to authenticate with DataCite. Please "
+            "contact support and/or open an issue at "
+            "https://github.com/Garden-AI/garden-backend/issues. "
+        )
+        logger.error(f"DATACITE_* environment variables not set. env: {os.environ}")
+        return {
+            "statusCode": 503,
```

Comment:
I think 503 usually means "we're temporarily down for maintenance." imo this is is a 500 because something about our service is messed up.

Reply:
good catch, thanks